### DISCUSSION TITLE
[Bug?] `npm i` doesn't install `mysql` package

### DISCUSSION THREAD

**Discussion by Master-Guy** [npm/feedback#986] [General]
Currently I'm trying to expand my NodeJS knowledge with some database access, and it's going pretty good.
Now I'm trying to get my package running, and as I usually do, I commit the code (excluding the node_modules folder and some other files) to Github, and download it on another system. There I run `npm i --omit=dev` and `npm run` to start my application.
Even though `npm i --omit=dev` reports no failures or issues, `npm run` fails due to the `mysql` package missing, even though it is listed in package.json and package-lock.json. Also I can see a folder called `node_modules/mysql`, which further adds to the confusion..
```
"dependencies": {
	"mysql": "^2.18.1",
	[...]
},
"devDependencies": {
	"@types/mysql": "^2.15.21",
	[...]
}
```
When I run `npm i` without `--omit=dev` it somehow remembers to install the `mysql` package that I'm missing before, showing: `added 1 package`
What could be the cause of this? The same happens with `npm ci`.

**Comment by Master-Guy** [comment]
Doing a file compare between node_modules with and without, I see these two differences:

1) `package-lock.json` contains an entry for `@types/mysql`:
```
		"node_modules/@types/mysql": {
			"version": "2.15.21",
			"resolved": "https://registry.npmjs.org/@types/mysql/-/mysql-2.15.21.tgz",
			"integrity": "sha512-NPotx5CVful7yB+qZbWtXL2fA4e7aEHkihHLjklc6ID8aq7bhguHgeIoC1EmSNTAuCgI6ZXrjt2ZSaXnYX0EUg==",
			"dev": true,
			"dependencies": {
				"@types/node": "*"
			}
		},
```
2) `@types/mysql` is added

There is however, not a single other difference..

**Comment by MylesBorins** [comment]
We don't offer general support here.
